### ISSUE TITLE
How to recognize "text content"

### ISSUE THREAD

**Issue by polterguy** [jstedfast/MimeKit#123]
Howdy, what is the best way to recognize "text content". Currently I am checking the 'Content-Type' of my Mime entities, for then to handle everything having a MediaType of "text" as text. The problem as I see it, is that this doesn't recognize everything that actually is text ...

For instance, "application/json" and "application/javascript", could also be treated like "text". Other examples includes "application/pgp-signature", which actually is "text content" ...

I need some mechanism, or algorithm, or logic, to be able to recognize everything which I internally within my system can safely store in a "string value".

What would be the best approach here ...?

Basically, to condense; Is there some sort of logic I could use to recognize every MimeEntity that I can safely store in a System.String, without messing up the actual content of the entity ...?

Sorry if this is off-topic, or a place where I really should have dived into the RFCs for MIME myself, but I assume I'm probably not the only one having this problem, hence I'm hoping for some solution/method inside MimeKit that could help me with this ...

Or at least, some general algorithm I could apply, to figure it out for myself ... :)

**Comment by jstedfast** [comment]
The only thing I can think of is to scan the decoded content for 0-bytes, if it has any of those, it's not text. But that's not a guarantee, it's just a "pretty good guess". I would say that if the data is completely in the 7-bit ASCII range, then there's a pretty high chance that it can be treated as text. The problem is the 8-bit range - it could be text *or* binary.

As sucky as it is, it might be better to simply have a registry of mime-types that you want to be able to treat as plain text.

**Comment by polterguy** [comment]
Darn it, thought so ... :(

Do you know about any good resources where I can figure out which types are the most common "text-based MIME types" ...?

**Comment by jstedfast** [comment]
No, but I do know of the registered mime-types list: http://www.iana.org/assignments/media-types/media-types.xhtml

**Comment by polterguy** [comment]
Thx, bro, I'll check it out :)
